### PR TITLE
provider/aws: support ec2 termination protection

### LINKED ISSUE #1233: Support for EC2 termination protection option
I was wondering if it would be possible to add termination protection to the aws_instance resource?

I saw this referenced briefly in https://github.com/hashicorp/terraform/issues/28 but figured I would open another issue to track it separately.

### PR DESCRIPTION
**Pull request by phinze**
closes #1233

### CODE REVIEW

**Comment by mitchellh** [comment]
Weird double negative weirdness here. I see its the API field name... but ugh. 

LGTM

File: builtin/providers/aws/resource_aws_instance_test.go [hunk 1]
```diff
@@ -629,6 +674,27 @@ resource "aws_instance" "foo" {
 }
 `
 
+func testAccInstanceConfigDisableAPITermination(val bool) string {
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_instance_test.go]: I like this style

**Comment by phinze** [comment]
Yeah agreed, decided to take the "throw up your hands and map it through" approach and link the AWS docs right away. I figure trying to make it clearer at TF layer will only cause sad edge cases of major confusion.

**Comment by phinze** [comment]
Failure is unrelated, merging through it so I don't leave this dangling.

File: builtin/providers/aws/resource_aws_instance_test.go [hunk 2]
```diff
@@ -629,6 +674,27 @@ resource "aws_instance" "foo" {
...
+		cidr_block = "10.1.1.0/24"
+		vpc_id = "${aws_vpc.foo.id}"
+	}
+
+	resource "aws_instance" "foo" {
+		# us-west-2
+		ami = "ami-4fccb37f"
+		instance_type = "m1.small"
+		subnet_id = "${aws_subnet.foo.id}"
+		disable_api_termination = %t
```
Comment (SydOps) [inline builtin/providers/aws/resource_aws_instance_test.go]: How to understand `%t` in this line?

Comment (phinze) [inline builtin/providers/aws/resource_aws_instance_test.go]: This is just boolean interpolation in [`Sprintf`](https://golang.org/pkg/fmt/) :+1:

**Pull request merged**